tests/test_common.py

```diff
@@ -0,0 +1,245 @@
+import dataclasses
+
+import pytest
+
+from genieutils.common import GenieClass, ByteHandler
+from genieutils.versions import Version
+
+class TestGenieClass:
+    @pytest.fixture
+    def genie_class(self):
+        return GenieClass()
+    
+    def test_from_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes(ByteHandler(b''))
+
+    def test_from_bytes_with_count(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.from_bytes_with_count(ByteHandler(b''), 0)
+
+    def test_to_bytes(self, genie_class: GenieClass):
+        with pytest.raises(NotImplementedError):
+            genie_class.to_bytes(Version.UNDEFINED)
+
+    def test_write_debug_string(self, genie_class: GenieClass):
+        assert genie_class.write_debug_string("foobar") == b'`\x0A\x06\x00foobar'
```

Comment:
This backtick looks a bit sus :eyes:

Reply:
Genuinely impressed you caught that. Fixed it up